tests/test/auto_traits.rs

```diff
@@ -217,3 +217,112 @@ fn enum_auto_trait() {
         }
     }
 }
+
+#[test]
+fn builtin_auto_trait() {
+    test! {
+        program {
+            #[auto] trait AutoTrait {}
+            struct Struct {}
+            enum Enum { Var1, Var2 }
+            fn func();
+
+            struct Marker {}
+            impl !AutoTrait for Marker {}
+
+            closure good_closure(self, arg: Marker) -> Marker { i32 }
+            closure bad_closure(self, arg: i32) -> i32  { Marker }
+
+            extern type Ext;
+            enum ExtEnum { GoodVariant, BadVariant(Ext) }
+        }
+
+        goal {
```

Comment:
Comments on these would be nice.

Can we also get a test of a tuple that *does* impl the auto trait.

Reply:
Oh, Haha, that's below.

Clever,  but I would feel better about splitting it up a bit.